### ISSUE TITLE
[CI] MlConfigIndexMappingsFullClusterRestartIT testMlConfigIndexMappingsAfterMigration fails in 7.x for BWC version < 6.6.0

### ISSUE THREAD

**Issue by droberts195** [elastic/elasticsearch#47286]
`MlConfigIndexMappingsFullClusterRestartIT.testMlConfigIndexMappingsAfterMigration` has failed in many BWC builds on the 7.x branch over the last few days.  A few examples are:

* https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+bwc/BWC_VERSION=6.3.0,nodes=centos-7&&immutable/167/console
* https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+bwc/BWC_VERSION=6.2.1,nodes=centos-7&&immutable/167/console
* https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+bwc/BWC_VERSION=6.5.3,nodes=centos-7&&immutable/167/console

But there are hundreds of others.

**Comment by droberts195** [comment]
Muted on the 7.x branch in a1d3711b5200d3bf4f374fa2b40ea60592dbfc51

**Comment by droberts195** [comment]
On closer inspection, all the failures occur when the old version in the BWC test is earlier than 6.6.0.

I am pretty sure that the problem lies in this setup code in `MlConfigIndexMappingsFullClusterRestartIT`:

```
    @Before
    public void waitForMlTemplates() throws Exception {
        List<String> templatesToWaitFor = XPackRestTestConstants.ML_POST_V660_TEMPLATES;
        XPackRestTestHelper.waitForTemplates(client(), templatesToWaitFor);
    }
```

It needs a change similar to the one made in https://github.com/elastic/elasticsearch/pull/47112/files#diff-2ce666ebd3338dbd6cc08ba88d6e86d2L71-R77 but with 6.6.0 instead of 7.4.0.

Previously when the BWC version was older than 6.6.0 an `AwaitBusy` in the test setup would never meet its objective but then carry on after 10 seconds (leading to a slowdown of 10 seconds in the test).  Now the `AssertBusy` fails the test in this case.

#47112 has exposed this.

**Comment by droberts195** [comment]
Fixed by #47289
